programs/gateway/src/instructions/update_voter_weight_record.rs

```diff
@@ -12,12 +15,18 @@ pub struct UpdateVoterWeightRecord<'info> {
     /// The Gateway Registrar
     pub registrar: Account<'info, Registrar>,
 
-    /// A gateway token from the gatekeeper network in the registrar.
+    /// An account that is either of type TokenOwnerRecordV2 or VoterWeightRecord
+    /// depending on whether the registrar includes a predecessor or not
+    /// CHECK: Checked in the code depending on the registrar
+    #[account()]
+    pub input_voting_weight: UncheckedAccount<'info>,
```

Comment:
I'm thinking maybe it should always be `VoterWeightRecord` to simplify plugins code.
We could expose `VoterWeightRecord` from `spl-governance` instead of using `TokenOwnerRecord`